src/libsystemd/sd-device/sd-device.c

```diff
@@ -1924,8 +1924,7 @@ _public_ int sd_device_get_sysattr_value(sd_device *device, const char *sysattr,
                         return r;
 
                 /* drop trailing newlines */
-                while (size > 0 && value[--size] == '\n')
```

Comment:
given you are not using `size` anymore here, you can remove the variable and pass NULL as las arg to read_full_virtual_file()

Reply:
Ouch.